src/pages/transactions/create/Create.tsx

```diff
@@ -0,0 +1,209 @@
+import { useTitle } from 'common/hooks/useTitle';
+import { date, endpoint } from 'common/helpers';
+import { TransactionInput } from 'common/interfaces/transactions';
+import { Container } from 'components/Container';
+import { DebouncedCombobox } from 'components/forms/DebouncedCombobox';
+import { Default } from 'components/layouts/Default';
+import { FormEvent, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { TransactionValidation } from '../common/validation/ValidationInterface';
+import { request } from 'common/helpers/request';
+import { useNavigate } from 'react-router-dom';
+import { route } from 'common/helpers/route';
+import { toast } from 'common/helpers/toast/toast';
+import { AxiosError } from 'axios';
+import { useCurrentCompany } from 'common/hooks/useCurrentCompany';
+import { DecimalNumberInput } from 'components/forms/DecimalNumberInput';
+import { useResolveCurrency } from 'common/hooks/useResolveCurrency';
+import { DecimalInputSeparators } from 'common/interfaces/decimal-number-input-separators';
+
+export function Create() {
+  const { t } = useTranslation();
+
+  const navigate = useNavigate();
+
+  const currencies = useCurrencies();
+
+  const company = useCurrentCompany();
+
+  const resolveCurrency = useResolveCurrency();
+
+  const { documentTitle } = useTitle('new_transaction');
+
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+
+  const [currencySeparators, setCurrencySeparators] = useState<
+    DecimalInputSeparators | undefined
+  >();
+
+  const [errors, setErrors] = useState<TransactionValidation>();
+
+  const [transaction, setTransaction] = useState<TransactionInput>();
+
+  const pages = [
+    { name: t('transactions'), href: '/transactions' },
+    { name: t('new_transaction'), href: '/transactions/create' },
+  ];
+
+  const getCurrencySeparators = (currencyId: string) => {
+    const currency = resolveCurrency(currencyId) || currencies[0];
+    return {
+      decimalSeparator: currency?.decimal_separator,
+      precision: currency?.precision,
+      thousandSeparator: currency?.thousand_separator,
+    };
+  };
+
+  const handleChange = (
+    property: keyof TransactionInput,
+    value: TransactionInput[keyof TransactionInput]
+  ) => {
+    if (property === 'currency_id') {
+      setCurrencySeparators(getCurrencySeparators(value?.toString() || ''));
+    }
+
+    setTransaction((prevState) => ({ ...prevState, [property]: value }));
+  };
+
+  const onSave = async (event: FormEvent<HTMLFormElement>) => {
+    event?.preventDefault();
+
+    setErrors(undefined);
+
+    setIsSaving(true);
+
+    const toastId = toast.processing();
+
+    try {
+      await request('POST', endpoint('/api/v1/bank_transactions'), {
+        ...transaction,
+        amount: Number(transaction?.amount),
+        base_type: transaction?.base_type === 'deposit' ? 'CREDIT' : 'DEBIT',
+      });
+      toast.success(t('created_transaction'), { id: toastId });
```

Comment:
This is not needed. Toast helpers exports single global instance, so carrying ID around isn't required.